pkg/apis/apis/v1alpha1/types_apiexport.go

```diff
@@ -220,24 +220,34 @@ type PermissionClaim struct {
 	IdentityHash string `json:"identityHash,omitempty"`
 }
 
-// +kubebuilder:validation:XValidation:rule="has(self.__namespace__) || has(self.name)",message="at least one field must be set"
+const (
+	ResourceSelectorAll = "*"
+)
+
+// ResourceSelector identifies the objects to be included within a PermissionClaim either by name and/or namespace.
+//
+// +kubebuilder:validation:XValidation:rule="has(self.namespaces) || has(self.names)",message="at least one field must be set"
 type ResourceSelector struct {
-	// name of an object within a claimed group/resource.
+	// names is a list of specific resources to select.
 	// It matches the metadata.name field of the underlying object.
-	// If namespace is unset, all objects matching that name will be claimed.
+	// An entry of "*" anywhere in the list means all object names of the group/resource within the "namespaces" list are claimed.
 	//
 	// +optional
-	// +kubebuilder:validation:Pattern="^([a-z0-9][-a-z0-9_.]*)?[a-z0-9]$"
-	// +kubebuilder:validation:MaxLength=253
-	// +kubebuilder:validation:MinLength=1
-	Name string `json:"name,omitempty"`
+	// +listType=set
+	// +kubebuilder:default={"*"}
+	Names []string `json:"names,omitempty"`
+
+	// TODO(nrb): this validation rule prevents the server from starting up due to cost estimates. Removed in order to allow startup until it's fixed
+	// kubebuilder:validation:XValidation:rule="self.all(x, x != \"\")",message="\"\" is not a valid namespace. Leave field blank for cluster-scoped resources"
```

Comment:
why so complicated? Why not just put it onto the namespace strings?

Reply:
How does one validate members of the list? I was unable to find this in any documentation.

Reply:
I type aliased `string` to get the validation. I was trying to avoid that solution, but I don't think it can be helped.